src/utils/proxy.js

```diff
@@ -185,7 +185,7 @@ const serveRedirect = async function ({ match, options, proxy, req, res }) {
 
   const staticFile = await getStatic(decodeURIComponent(reqUrl.pathname), options.publicFolder)
   if (staticFile) {
-    req.url = encodeURIComponent(staticFile) + reqUrl.search
+    req.url = encodeURI(staticFile) + reqUrl.search
```

Comment:
This fixes a problem where static URLs are escaped to `%2Findex.html`. Introduced in https://github.com/netlify/cli/pull/2665 and the tests still works.